File: Swift/프로그래머스/1/118666. 성격 유형 검사하기/성격 유형 검사하기.swift
import Foundation

func solution(_ survey:[String], _ choices:[Int]) -> String {
    var map: [String: Int] = [:]
    
    for i in survey.indices {
        var choice = choices[i]
        
        if choice > 0 && choice < 4 {
            let key = String(survey[i].first!)
            let score = 4 - choice
            map[key, default: 0] += score
        }
        else {
            let key = String(survey[i].last!)
            let score = choice - 4
            map[key, default: 0] += score
        }
    }
    let rt = map["R", default: 0] >= map["T", default: 0] ? "R" : "T"
    let cf = map["C", default: 0] >= map["F", default: 0] ? "C" : "F"
    let jm = map["J", default: 0] >= map["M", default: 0] ? "J" : "M"
    let an = map["A", default: 0] >= map["N", default: 0] ? "A" : "N"
    
    let result = rt + cf + jm + an
    print(result)
    return result
    
}

File: Java/프로그래머스/1/118666. 성격 유형 검사하기/성격 유형 검사하기.java
import java.util.HashMap;
import java.util.HashSet;
import java.util.*;
class Solution {
    public String solution(String[] survey, int[] choices) {
        HashMap<Character,Integer> map = new HashMap<>();

        for(int i=0;i<survey.length;i++){
            int value = choices[i]; /// 5

            if(value>0&& value<4){
                char suv = survey[i].charAt(0); //A
                map.put(suv,map.getOrDefault(suv,0)+4-value);
            }
            else{
                char suv2 = survey[i].charAt(1); //A
                map.put(suv2,map.getOrDefault(suv2,0)+value-4);
            }
        }
        return new StringBuilder()
                .append(map.getOrDefault('R',0)>=map.getOrDefault('T',0)?'R':'T')
                .append(map.getOrDefault('C',0)>=map.getOrDefault('F',0)?'C':'F')
                .append(map.getOrDefault('J',0)>=map.getOrDefault('M',0)?'J':'M')
                .append(map.getOrDefault('A',0)>=map.getOrDefault('N',0)?'A':'N')
                .toString();
    }
}